### PR TITLE
Use `default_client` in `get_scheduler`

### PR DESCRIPTION
**Pull request by mrocklin**
See https://github.com/dask/distributed/issues/7377

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
Thanks @mrocklin! Is this something we can add a test for? Clearly our current test suite isn't capturing the many-client-in-many-threads issues you ran into

Also, in https://github.com/dask/distributed/issues/7377 you mentioned `Client.current()` is using `ContextVars` and might be a better way to go. Should we be using `Client.current()` here instead?

**Review by fjetter — APPROVED** [review]

**Comment by fjetter** [comment]
> Also, in https://github.com/dask/distributed/issues/7377 you mentioned Client.current() is using ContextVars and might be a better way to go. Should we be using Client.current() here instead?

With https://github.com/dask/distributed/issues/7377 `Client.current` and `default_client` becomes the same thing. default_client chooses current first and then picks a global default. current falls back to global default if there is no current.

File: dask/base.py [hunk 1]
```diff
@@ -1370,9 +1370,12 @@ def get_scheduler(get=None, scheduler=None, collections=None, cls=None):
...
                 return named_schedulers[scheduler]
             elif scheduler in ("dask.distributed", "distributed"):
-                from distributed.worker import get_client
+                from dask.distributed import default_client, get_client
 
-                return get_client().get
+                try:
+                    return default_client().get
+                except ValueError:
+                    return get_client().get
```
[the remark this answers is not in the document]
Comment (fjetter) [reply dask/base.py:1378]: get_client will actually use the default client by default

get_clients resolution is

1. If we are a worker, select the worker and call Worker._get_client
2. Worker._get_client tries to get a default client first https://github.com/dask/distributed/blob/19deee3b1ac7298dd6b7319481417b0b4e1986d8/distributed/worker.py#L2530-L2534
3. If a ValueError is raised, it instantiates a new one.

This try/except should not be necessary and default_client should directly do the right thing, even without https://github.com/dask/distributed/issues/7377

**Review by fjetter — CHANGES_REQUESTED** [review]

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the extra context @fjetter. Just pushed a commit to make this PR just swap out `get_client()` for `default_client()`

**Comment by fjetter** [comment]
> Thanks for the extra context @fjetter. Just pushed a commit to make this PR just swap out get_client() for default_client()

No, I think we need `get_client` here, i.e. I suggest to simply close this PR and think `main` is as good as it gets.

Specifically, `get_client` will be able to open and create a client if the task is running on a worker and there is no default client registered, yet. 
`default_client` will just raise if there is no other

Effectively

`get_client` > `default_client == Client.current`

**Comment by fjetter** [comment]
Practically, it won't matter much. This code branch is typically triggered because somebody set a default client. Still, get_client should "always" work and defaults to the default_client anyhow

**Pull request closed without merging**